phoenix-core-client/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -941,12 +940,7 @@ private ScansWithRegionLocations getParallelScans(byte[] startKey, byte[] stopKe
             Scan scanFromContext = context.getScan();
             if (scanRanges.getPointLookupCount() == 1) {
                 // leverage bloom filter for single key point lookup by turning scan to Get Scan#isGetScan()
-                try {
-                    scanFromContext = new Scan(context.getScan());
-                } catch (IOException e) {
-                    throw new PhoenixIOException(e);
-                }
-                scanFromContext.withStopRow(scanFromContext.getStartRow(), scanFromContext.includeStartRow());
+                scanFromContext.withStopRow(scanFromContext.getStartRow(), true);
```

Comment:
I actually didn't want to change the scan stored in StatementContext. I wanted to modify the scan we generate in getIterators.

Reply:
You generate a scan in StatementContext and then we generate the actual scans when we are creating the iterators like a high level scan in statementcontext could generate multiple parallel scans one for each region server. That is why I was copying the original scan stored in StatementContext.

Reply:
I see, then we can revert it.

Reply:
Earlier the scan object had stop key as next(start key) for point lookups. Take a look at this test case https://github.com/apache/phoenix/blob/master/phoenix-core/src/test/java/org/apache/phoenix/compile/WhereOptimizerTest.java#L988-L992